python/dolfinx/plot.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2021 Jørgen S. Dokken and Garth N. Wells
+#
+# This file is part of DOLFINX (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+"""Support function for plotting"""
+
+import functools
+
+import numpy as np
+
+from dolfinx import cpp, fem
+
+# Permutation for Dolfinx DG layout to VTK
+# Note that third order tetrahedrons has a special ordering: https://gitlab.kitware.com/vtk/vtk/-/issues/17746
+_perm_dg = {cpp.mesh.CellType.triangle: {1: [0, 1, 2], 2: [0, 2, 5, 1, 4, 3], 3: [0, 3, 9, 1, 2, 6, 8, 7, 4, 5],
+                                         4: [0, 4, 14, 1, 2, 3, 8, 11, 13, 12, 9, 5, 6, 7, 10]},
+            cpp.mesh.CellType.tetrahedron: {1: [0, 1, 2, 3], 2: [0, 2, 5, 9, 1, 4, 5, 6, 7, 8],
+                                            3: [0, 3, 9, 19, 1, 2, 6, 8, 7, 4, 10, 16, 12, 17, 15, 18, 11, 14, 13, 5]}}
+_perm_dq = {cpp.mesh.CellType.quadrilateral: {1: [0, 1, 3, 2], 2: [0, 2, 8, 6, 1, 5, 7, 3, 4],
+                                              3: [0, 3, 15, 12, 1, 2, 7, 11, 13, 14, 4, 8, 5, 6, 9, 10]},
+            cpp.mesh.CellType.hexahedron: {1: [0, 1, 3, 2, 4, 5, 7, 6],
+                                           2: [0, 2, 8, 6, 18, 20, 26, 24, 1, 5, 7, 3, 19,
+                                               23, 25, 21, 9, 11, 17, 15, 12, 14, 10, 16, 4, 22, 14]}}
+
+# NOTE: Edge visualization of higher order elements are sketchy, see:
+# https://github.com/pyvista/pyvista/issues/947
+
+
+def _transpose(map):
+    """Transpose of the map. E.g., is `map = [1, 2, 3, 0]`, the
+    transpose will be `[3 , 0, 1, 2 ]`.
+
+    """
+    transpose = np.zeros(len(map), dtype=np.int32)
+    for i in range(len(map)):
+        transpose[map[i]] = i
+    return transpose
+
+
+@functools.singledispatch
+def create_pyvista_topology(mesh: cpp.mesh.Mesh, dim: int, entities=None):
```

Comment:
Could this be called `create_vtk_topology`? It uses VTK ordering, produces pure arrays, has nothing to do with pyvista per se.